sqlglot/dialects/dialect.py

```diff
@@ -361,7 +361,7 @@ def create_with_partitions_sql(self: Generator, expression: exp.Create) -> str:
         expression = expression.copy()
         prop = expression.find(exp.PartitionedByProperty)
         this = prop and prop.this
-        if prop and not isinstance(this, exp.Schema):
+        if prop and this and not isinstance(this, exp.Schema):
```

Comment:
handle `this` being `None`

Reply:
this one is not necessary, line 363 indicates that this is not None 

this = prop and prop.this, so this must not be None

Reply:
`prop.this` can return `None` though

Reply:
this code is a bit confusing in general

maybe you should remove line 363

and do

if prop and not isinstance(prop, exp.Schema)

Reply:
👍 good point, i've removed line 363, and updated the line below that used `this` ... I still need to check `prop.this` is not `None` before its used below... have a look at the latest commit and see what you think